packages/location-management/src/components/LocationTagAddEdit/Form.tsx

```diff
@@ -0,0 +1,163 @@
+const offsetLayout = { wrapperCol: { offset: 8, span: 11 } };
+
+const status = [
+  { label: 'Active', value: true },
+  { label: 'Inactive', value: false },
+];
+
+interface FormField {
+  name: string;
+  description: string;
+  active: boolean;
+}
+
+/** yup validations for practitioner data object from form */
+const userSchema = Yup.object().shape({
+  name: Yup.string().typeError('Name must be a String').required('Name is Required'),
+  active: Yup.boolean().typeError('Status must be a Boolean').required('Status is Required'),
+  description: Yup.string()
+    .typeError('Description must be a String')
+    .required('Description is Required'),
+});
+
+interface Props {
+  id?: string;
+}
+
+/**
+ * Handle form submission
+ *
+ * @param {Object} values the form fields
+ * @param {string} accessToken api access token
+ * @param {object} props component props
+ * @param {Function} setSubmitting method to set submission status
+ */
+export const onSubmit = (
+  values: FormField,
+  accessToken: string,
+  props: Props,
+  setSubmitting: (isSubmitting: boolean) => void
+) => {
+  const serve = new OpenSRPService(accessToken, API_BASE_URL, LOCATION_TAG_ALL);
+
+  const payload: LocationTagPayloadPOST | LocationTagPayloadPUT = values;
+
+  if (props.id) {
+    (payload as LocationTagPayloadPUT).id = props.id;
+    serve
+      .update(payload)
+      .then(() => {
+        notification.success({ message: 'Location Tag updated successfully', description: '' });
+        setSubmitting(false);
+        history.goBack();
+      })
+      .catch((e: Error) => {
+        notification.error({ message: `${e}`, description: '' });
+        setSubmitting(false);
+      });
+  } else {
+    serve
+      .create(payload)
+      .then(() => {
+        notification.success({ message: 'Location Tag successfully', description: '' });
+        setSubmitting(false);
+        history.goBack();
+      })
+      .catch((e: Error) => {
+        notification.error({ message: `${e}`, description: '' });
+        setSubmitting(false);
+      });
+  }
+};
+
+export const Form: React.FC<Props> = (props: Props) => {
+  const accessToken = useSelector((state) => getAccessToken(state) as string);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [initialValue, setInitialValue] = useState<FormField>({
+    name: '',
+    description: '',
+    active: true,
+  });
+
+  useEffect(() => {
+    if (isLoading) {
+      if (props.id) {
+        const serve = new OpenSRPService(accessToken, API_BASE_URL, LOCATION_TAG_GET + props.id);
+        serve
+          .list()
+          .then((response: LocationTag) => {
+            setInitialValue({
+              active: response.active,
+              description: response.description,
+              name: response.name,
+            });
+            setIsLoading(false);
+          })
+          .catch((e) => notification.error({ message: `${e}`, description: '' }));
```

Comment:
Seems when the API fails isloading will be forever true thus showing unending loader

Reply:
continuing this conversation on latest review by kelvin